### PR TITLE
Override the birth log asset label and description

### PR DESCRIPTION
**Pull request by mstenta**
This overrides the "Asset" field on Birth logs to change the label and description as follows.

- Label: Children
- Description: Which child assets were born?

See @paul121's comment here for the inspiration: https://github.com/farmOS/farmOS/pull/774#issuecomment-1894813820

This is nearly identical in implementation to #774 

I'm open to suggestions on the label/description. I thought "Children" would be better than "Child" in this case, even though it's inconsistent with our normally singular labels. I don't feel too strongly though, so if others prefer "Child" I can change it to that.

### CODE REVIEW

**Comment by mstenta** [comment]
Oops one of the birth log tests is failing... I'm guessing the additional config requires something special in the kernel test. Standby for a fix...

**Comment by mstenta** [comment]
Oh just realized we can also remove all of this: https://github.com/farmOS/farmOS/blob/0903f96b2bf111c9c0f2624af28e4a2e00b47d19/modules/log/birth/farm_birth.module#L14-L42

File: modules/log/birth/farm_birth.post_update.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Override the birth log asset label and description.
+ */
+function farm_birth_post_update_override_birth_asset_label_description(&$sandbox) {
+
+  // Override the asset field label and description on birth logs to make it
+  // clear that it should be used to reference the child assets that were born.
+  $config = \Drupal::service('entity_field.manager')->getBaseFieldDefinitions('log')['asset']->getConfig('birth');
+  $config->set('label', 'Children');
+  $config->set('description', 'Which child assets were born?');
+  $config->save();
```
Comment (paul121) [inline modules/log/birth/farm_birth.post_update.php]: Oh hmm I would have guessed that we create a new `base_field_override` config entity. This must be happening behind the scenes. But is it possible to do it more directly?

Comment (mstenta) [reply modules/log/birth/farm_birth.post_update.php]: Ah good question. That may be possible too. :thinking: 

My code was based on this: https://www.previousnext.com.au/blog/overriding-base-field-labels-and-descriptions-of-drupal-entities-without-custom-code

I'm not sure what the "right" / "best" way to do it is. Perhaps there's an argument that going through `entity_field.manager` has some benefits because it is doing it from a higher level? It probably doesn't make a big difference either way. The nice thing about this is that we only need to set `label` and `description`. Constructing a `BaseFieldOverride` entity would require defining *all* the properties of that entity, which we get for free via `getConfig('birth')`.

**Comment by mstenta** [comment]
> Oops one of the birth log tests is failing... I'm guessing the additional config requires something special in the kernel test. Standby for a fix...

Interesting... I traced this to a Drupal core issue: https://www.drupal.org/project/drupal/issues/3193351

Tl;dr: If you have a base field that also has a validation constraint associated with it, and then you provide a base field override for it, the constraint is no longer applied.

There is a simple workaround: use `hook_entity_bundle_field_info()` instead of `hook_entity_field_info_alter()` to add the constraint, and load the `BaseFieldOverride` first.

I've updated our code and included a comment that points to the core issue as the reasoning. Force pushed... tests should pass now.

**Comment by mstenta** [comment]
Force pushed to check `$config->isNew()` in the update hook, in case someone is already overriding this config.

**Review by paul121 — APPROVED** [review]
